### PR TITLE
GEOPY-760: Create spatial tiling estimator

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-760 - Create spatial tiling estimator**

### CODE REVIEW

File: geoapps_utils/utils/numerical.py [hunk 1]
```diff
@@ -134,3 +134,44 @@ def weighted_average(  # pylint: disable=too-many-locals
...
+    """
+    a, b = 0, 1
+    series = []
+    while a < n:
+        series.append(a)
+        a, b = b, a + b
+    return np.array(series)
+
+
+def fit_circle(x_val: np.ndarray, y_val) -> tuple:
```
Comment (benk-mira) [inline geoapps_utils/utils/numerical.py:155]:
```suggestion
def fit_circle(x_val: np.ndarray, y_val) -> tuple[float, np.ndarray, np.ndarray]:
```

File: geoapps_utils/utils/numerical.py [hunk 2]
```diff
@@ -134,3 +134,44 @@ def weighted_average(  # pylint: disable=too-many-locals
...
+    """
+    Compute the least-square circle fit to a set of points.
+
+    :param x_val: x-coordinates of the points
+    :param y_val: y-coordinates of the points
+
+    :return: radius, x0, y0
```
Comment (benk-mira) [inline geoapps_utils/utils/numerical.py:162]: docstring these individually:
```
:return radius: blah blah.
:return x0: blah blah.
:return y0: and on and on.
```

**Review by benk-mira — CHANGES_REQUESTED** [review]
Looks good

File: geoapps_utils/utils/numerical.py [hunk 3]
```diff
@@ -134,3 +134,44 @@ def weighted_average(  # pylint: disable=too-many-locals
...
+
+    # Right-hand side
+    rhs = np.zeros((len(x_val), 1))
+    rhs[:, 0] = (x_val * x_val) + (y_val * y_val)
+
+    # Find the least-square solution
+    coef, _, _, _ = np.linalg.lstsq(lin_eqs, rhs, rcond=None)
+
+    # Compute radius
+    radius = (coef[0] ** 2.0 + coef[1] ** 2.0 + coef[2]) ** 0.5
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps_utils/utils/numerical.py:175]: No the last coefficient represents x0**2 + y0**2 - r**2

[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps_utils/utils/numerical.py:175]: I added notes in the docstrings

**Review by benk-mira — APPROVED** [review]
thanks for the docstring updates

**Pull request merged**